### PR TITLE
Smart content provider: Added sorting

### PR DESCRIPTION
**Pull request by trickreich**
| Q | A
| --- | ---
| Bug fix? | no
| New feature? | yes
| BC breaks? | no
| Deprecations? | no
| Related issues/PRs | #62 #63 
| License | MIT
| Documentation PR | --

#### What's in this PR?

Added sorting to smart content provider.

### CODE REVIEW

**Comment by trickreich** [comment]
@wachterjohannes Is implemented, please have a look :)

File: Tests/Functional/Content/ArticleDataProviderTest.php [hunk 1]
```diff
@@ -59,7 +59,7 @@ public function testResolveDataItemsTypeParam()
 
         $result = $dataProvider->resolveDataItems(
             [],
-            ['type' => new PropertyParameter('type', 'video')],
+            ['types' => new PropertyParameter('types', 'video')],
```
Comment (wachterjohannes) [inline Tests/Functional/Content/ArticleDataProviderTest.php]: can you add a test the filter with a single and multiple types?

File: Tests/Functional/Content/ArticleDataProviderTest.php [hunk 2]
```diff
@@ -65,7 +66,34 @@ public function testResolveDataItemsTypeParam()
...
+        $result = $dataProvider->resolveDataItems(
+            [],
+            ['types' => new PropertyParameter('types', 'other')],
+            ['locale' => 'de']
+        );
+        $this->assertInstanceOf(DataProviderResult::class, $result);
+        $this->assertCount(0, $result->getItems());
+
+        // get all articles with types video and blog
+        $result = $dataProvider->resolveDataItems(
```
Comment (wachterjohannes) [inline Tests/Functional/Content/ArticleDataProviderTest.php]: move it to an own test method. else ACK

**Comment by wachterjohannes** [comment]
@trickreich it works. can you split the testcase and we can merge this.

**Comment by trickreich** [comment]
@wachterjohannes Yeah, merge this ;)

**Pull request merged**